apps/cache-manager/src/utils.ts

```diff
@@ -0,0 +1,122 @@
+import {GetTableEntityResponse, TableClient, TableEntityResult, TableServiceClient} from '@azure/data-tables'
+import axios from 'axios'
+
+import {UpsertTableItem} from './types'
+
+export class AzureTableStorage {
+  static connectTableClient = async (connectionString: string, tableName: string) => {
+    let client
+    try {
+      client = TableClient.fromConnectionString(connectionString, tableName)
+    } catch (error) {
+      console.log(`[connectTableClient] The following error occurred ${error}`)
+    }
+
+    return client
+  }
+
+  static createTable = async (connectionString: string, tableName: string) => {
+    let result
+    try {
+      const tableServiceClient = await TableServiceClient.fromConnectionString(connectionString)
+      await tableServiceClient.createTable(tableName)
+
+      const task = {
+        partitionKey: 'federation',
+        rowKey: 'header',
+        description: 'Header module',
+        value: '--[]--header--[]--',
+      }
+
+      const tableClient = await this.connectTableClient(connectionString, tableName)
+
+      result = await tableClient?.createEntity(task)
+    } catch (error) {
+      console.log(`[createTable] The following error occurred: ${error}`)
+    }
+    return result
+  }
+
+  static deleteTable = async (connectionString: string, tableName: string) => {
+    let result
+    try {
+      const tableClient = await this.connectTableClient(connectionString, tableName)
+
+      result = await tableClient?.deleteTable()
+    } catch (error) {
+      console.log(`[deleteTable] The following error occurred: ${error}`)
+    }
+    return result
+  }
+
+  static getTableItem = async (client: TableClient | undefined, partitionKey: string, rowKey: string) => {
+    let tableItem: GetTableEntityResponse<TableEntityResult<object>> | undefined
+
+    try {
+      tableItem = await client?.getEntity(partitionKey, rowKey)
+    } catch (error) {
+      console.log(`[getTableItem] The following error occurred: ${error}`)
+    }
+
+    return tableItem
+  }
+
+  static upsertTableItem = async (client: TableClient | undefined, item: UpsertTableItem) => {
+    let tableItem
+    try {
+      tableItem = await client?.upsertEntity(item)
+    } catch (error) {
+      console.log(`[upsertTableItem] The following error occurred: ${error}`)
+    }
+
+    return tableItem
+  }
+
+  static deleteTableItem = async (client: TableClient | undefined, partitionKey: string, rowKey: string) => {
+    let tableItem
+    try {
+      tableItem = await client?.deleteEntity(partitionKey, rowKey)
+    } catch (error) {
+      console.log(`[deleteTableItem] The following error occurred: ${error}`)
+    }
+
+    return tableItem
+  }
+}
+
+export async function getComponent(data: object, port: string) {
+  const res = await axios(`http://localhost:${port}/prerender`, {
+    method: 'POST',
+    data,
+    headers: {
+      'content-type': 'application/json',
```

Comment:
Do we need to forward on headers here?

Reply:
I don't think there are anymore required headers for this call, but might be worth us passing the request headers through in case they're needed going forward

Reply:
Done everything else, i think there's a fair bit of overlap with some of the other stuff Mete is working on, so we ok to park this as non MVP and merge the rest with this on the backlog?